--- a/src/upnpcommands.c
+++ b/src/upnpcommands.c
@@ -357,7 +357,7 @@ UPNP_AddPortMapping(const char * controlURL, const char * servicetype,
 		{"NewInternalPort", inPort},
 		{"NewInternalClient", inClient},
 		{"NewEnabled", "1"},
-		{"NewPortMappingDescription", desc?desc:"libminiupnpc"},
+		{"NewPortMappingDescription", desc?desc:"miniupnpc"},
 		{"NewLeaseDuration", leaseDuration?leaseDuration:"0"},
 		{NULL, NULL}
 	};
