Signed-off-by: lean <chenlin@fnnas.com>

--- a/src/widgets/widgets/qcommandlinkbutton.cpp
+++ b/src/widgets/widgets/qcommandlinkbutton.cpp
@@ -11,7 +11,9 @@
 #include <qmath.h>
 
 #include "private/qpushbutton_p.h"
+#if QT_CONFIG(style_stylesheet)
 #include "private/qstylesheetstyle_p.h"
+#endif
 
 QT_BEGIN_NAMESPACE
 
@@ -285,11 +287,15 @@
         // - or it has been set in the style sheet, in which case we don't want to override it here.
         // When a style sheet with an icon is replaced by one without an icon, the old icon
         // will be reset, when baseStyle()->repolish() is called.
+#if QT_CONFIG(style_stylesheet)
         if (!qobject_cast<QStyleSheetStyle *>(style())) {
+#endif
             QStyleOptionButton opt;
             initStyleOption(&opt);
             setIcon(style()->standardIcon(QStyle::SP_CommandLink, &opt, this));
+#if QT_CONFIG(style_stylesheet)
         }
+#endif
     }
 
     return QPushButton::event(e);
