From 18c955901266b2899d11505763b156be7b202e99 Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Tue, 8 Mar 2022 16:09:40 +0100
Subject: [PATCH 5/5] auxv

---
 node/Utils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/node/Utils.cpp b/node/Utils.cpp
index a0bf0707..9d924238 100644
--- a/node/Utils.cpp
+++ b/node/Utils.cpp
@@ -29,7 +29,7 @@
 #include <sys/uio.h>
 #include <dirent.h>
 #ifdef ZT_ARCH_ARM_HAS_NEON
-#ifdef __LINUX__
+#if 0
 #include <sys/auxv.h>
 #endif
 #endif
-- 
2.35.1

